src/Backend/Modules/Authentication/Actions/ResetPassword.php

```diff
@@ -61,8 +61,8 @@ public function execute(): void
     private function isThePasswordResetKeyCorrect(): bool
     {
         // catch the key and e-mail address from GET
-        $this->email = urldecode(\SpoonFilter::getGetValue('email', null, ''));
-        $this->key = \SpoonFilter::getGetValue('key', null, '');
+        $this->email = urldecode($this->getRequest()->query->get('email', ''));
+        $this->key = $this->getRequest()->query->get('key', '');
 
         // if the email or the key aren't set, redirect the user
         if ($this->email !== '' && $this->key !== '') {
```

Comment:
you can change the !== to !empty($this->email) maybe?

but maybe you can clean this code up a bit better, the query has also the option to check if data is filled, maybe test what the has does with empty strings, cause in that case you would be able to run this check on the query of the request